src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
@rnveach 
> I say let's split this into new issue. This check is already complex enough and let's just get this PR done.

Sure. When the new issue is created, I am willing to continue to help with that. :)

> Unreachable statements won't change the fact that a variable can be `final` or not, right?

Yep.

> Our main concern is can the variable be declared `final` if it is compilable.

But the point is the unreachable statement would lead to a compile error. smth like this:
```
void foo() {
  int a;
  a = 1;
  return;
  a = 2; // compile error
}
```
We would get an `Unreachable statement` error. If I understood correctly, we don't need to concern about the cases like that since it is not compilable.
In fact, I haven't handle the unreachable cases in this commit by now. I assume it could be not easy if we need to concern about the unreachable statement with `return` and `break`.

> ... but false negatives wouldn't show there.

Maybe I don't understand clearly... Is there anything unexpected in the diff report? If the answer is yes, I will try to fix them, or split it into new issue?